ISSUE 384 add tests for java client metrics

kevinearls opened the pull request:
Signed-off-by: Kevin Earls <kearls@redhat.com>

jpkrohling commented on jaeger-micrometer/src/test/java/io/jaegertracing/micrometer/MicrometerTest.java:
```diff
@@ -85,4 +120,83 @@ public void testTimer() {
+              .withTag("foo", "bar" + i)
+              .start();
+
+      if (i % 3 == 0) {
```
Can you add a comment explaining what this is for?

jpkrohling replied:
I'm not quite sure I understand what is going on here: are you finishing only every third span? Would it cause any resource leaks?

kevinearls commented:
@jpkrohling I've changed the title and added a comment.  I'm not finishing those spans just so that finished spans and started spans counts will be different.  It's not really completely necessary, so I can remove it you think that would be better.

jpkrohling commented:
>  It's not really completely necessary, so I can remove it you think that would be better.

That's fine, your comment there should be enough. I'm just a bit concerned about possible side effects, but for now, let's just keep an eye on suspicious failures elsewhere.

yurishkuro commented:
@kevinearls a couple of stylistic comments about putting "ISSUE 384" into the title:
  * when the PR is merged, its title is usually used as the final commit message, but it does not match out commit message guidelines
  * putting issue reference in the title does not create any linkage in GitHub between the issue and the PR. The preferred way is to put "Resolves #384" in the PR description, which not only links them, but also auto-resolves the issue on PR merge.
